### PR TITLE
Task/70 implement invite command

### LINKED ISSUE #70: [TASK] Implement INVITE command
## Context
`InviteCommand.cpp` on `test/command-parsing` is a validated stub: it
enforces exactly 2 arguments in the order `<nick> <channel>`, validates
the nick via `requireWord` and the channel via `requireChannel`, then logs
to stdout. The full IRC logic — operator checks, invite list management,
and reply messages — is not yet wired pending Server/Client/Channel context.

Note the argument order: `INVITE <nick> <channel>` — nick is `args[0]`,
channel is `args[1]`. This is the correct RFC 1459 order and must be
preserved.

## Objective
Complete `InviteCommand::execute()` with full IRC-compliant behaviour:
operator privilege check on invite-only channels, target existence/presence
validation, invite list update, and the `RPL_INVITING` reply plus the
`INVITE` message forwarded to the target.

## Acceptation
- [ ] `InviteCommand::NAME` is defined as `"INVITE"` in the `.cpp`
- [ ] Argument validation (already in stub, must be kept):
  - Exactly 2 arguments (`<nick> <channel>`)
  - `args[0]` validated via `requireWord` (target nick)
  - `args[1]` validated via `requireChannel`
- [ ] Client must be fully registered
- [ ] Channel must exist; reply `ERR_NOSUCHCHANNEL (403)` if not
- [ ] Sender must be a member of the channel; reply `ERR_NOTONCHANNEL (442)`
  if not
- [ ] If channel has mode `+i`: sender must be a channel operator; reply
  `ERR_CHANOPRIVSNEEDED (482)` if not
- [ ] Target nick must exist on the server; reply `ERR_NOSUCHNICK (401)`
  if not
- [ ] Target must not already be in the channel; reply
  `ERR_USERONCHANNEL (443)` if already present
- [ ] On success:
  - Add target nick to the channel's invite list
  - Send `RPL_INVITING (341)` to the inviting client:
    `":server 341 <inviter> <target> <channel>"`
  - Forward `:<inviter>!<user>@<host> INVITE <target> <channel>` to the
    target client
- [ ] Compiles cleanly with `-Wall -Wextra -Werror -std=c++98`

## Dependencies
- #60 — `ACommand` validators in place
- #67 — JOIN and invite-only mode (`+i`) must be working for full test coverage
- #71 — MODE `+i` implementation needed to test the operator-only invite path
- Server/Client/Channel context objects must expose: channel map, mode flags,
  invite list, operator set check, nick map, reply sender

## Technical notes
- Current stub argument order is `requireWord(args, 0, "other nickname")`,
  `requireChannel(args, 1, "channel name")` — this is correct RFC order and
  must not be swapped
- `ERR_NOSUCHCHANNEL (403)`: `":server 403 <nick> <channel> :No such channel"`
- `ERR_NOTONCHANNEL (442)`: `":server 442 <nick> <channel> :You're not on that channel"`
- `ERR_CHANOPRIVSNEEDED (482)`: `":server 482 <nick> <channel> :You're not channel operator"`
- `ERR_NOSUCHNICK (401)`: `":server 401 <nick> <target> :No such nick/channel"`
- `ERR_USERONCHANNEL (443)`: `":server 443 <nick> <target> <channel> :is already on channel"`
- `RPL_INVITING (341)`: `":server 341 <inviter> <target> <channel>"`
- The invite list is per-channel and persists until the target joins or the
  channel is destroyed — it is consumed by `JoinCommand` when mode `+i` is active
- On non-invite-only channels, any member (not just operators) may invite

### PR DESCRIPTION
**Pull request by lcalero42**
## Summary
Implements `INVITE` command to be able to **invite an user on a channel we are in**.
`INVITE <nickname> <channel>`

## Why
This command is **needed by the subject** and make possible for our users to **invite any user in the server on the channel** we are in.

## Changes
- Implemented command error checks according to _IRC docs_
- Implemented reply to the target user whenever he gets invited
- Implemented reply to the source client to let him know he is inviting someone successfully

## Tests
- [x] Test with `irssi`,  try to invite someone that **does not exist in the server**
- [x] Test with `irssi`, try to invite someone **on a channel you are not in**
- [x] Test with `irssi`, try to invite someone **on a non-existing channel**
- [x] Test with `irssi`, try to invite someone **already in the channel specified**
- [x] test with `irssi`, try to invite someone **when `mode +i` is activated** (should not work when user is not operator)
- [x] Code compiles with `-Wall -Wextra -Werror -std=c++98`, no warnings
- [x] Manual test with `nc -C 127.0.0.1 6667` when relevant
- [x] Manual test with the reference IRC client when relevant
- [x] Edge cases tested (partial messages, malformed commands, client disconnects)

## Risks
Testing with `/mode +i` wasnt possible since join is not working properly, knowing that the fixes of join are present in the PR #123 (it would be better to validate this specified PR before this one).

## Linked issue
Closes #70 

---

## Merge policy

- Use squash or rebase merges only (no merge commits).
- Ensure dependancy are merged and there's no `blocked` label
- Do not merge if any CI check is red.
- Do not merge without at least one review approval.
- Ensure the branch is up to date with main before merging.

### CODE REVIEW

File: src/commands/operator/InviteCommand.cpp [hunk 1]
```diff
@@ -3,11 +3,67 @@
...
+	{
+		ServerReply::reply(client, ServerReply::ERR_NOSUCHCHANNEL, args[1]);
+		return;
+	}
+	if (!checkInvitePossible(client, *target, chan, args))
+		return;
+	ServerReply::reply(client, *chan, ServerReply::RPL_INVITING, args[0]);
+	target->reply(":" + client.getNickname() + " INVITE "
+				  + target->getNickname() + " " + chan->getName());
+}
```
Comment (ketodin) [inline src/commands/operator/InviteCommand.cpp:45]: There's no state change inside channel at all. Even if the `INVITE` command was correct and should work it currently only reply, it never adds the `Client` itself to the `Channel` invite list.

**Review by ketodin — CHANGES_REQUESTED** [review]
Needs #123 merged first to avoid dumb conflicts.
Overall good, although similar to requested changes in #123 the `Check...()` function inside the `INVITE` class should not take any pointers and should always use references.

**Review by eterrale — APPROVED** [review]

**Review by ketodin — APPROVED** [review]
